### PR TITLE
Tests: Bidding with royalties

### PR DESCRIPTION
**Pull request by nmboavida**

### CODE REVIEW

File: sources/standards/royalties/royalty.move [hunk 1]
```diff
@@ -115,6 +115,8 @@ module nft_protocol::royalty {
         amount: u64,
     ): u64 {
         if (!contains_royalty_strategy<BpsRoyaltyStrategy>(domain)) {
+            // TODO: This is dangerous because it can lead to
```
Comment (bausano) [inline sources/standards/royalties/royalty.move:118]: can we create a new issue with some proposed solutions and _why_ it's dangerous

File: sources/standards/royalties/royalty_strategy.move [hunk 2]
```diff
@@ -11,7 +11,9 @@ module nft_protocol::royalty_strategy_bps {
     }
 
     public fun calculate(domain: &BpsRoyaltyStrategy, amount: u64): u64  {
-        amount / BPS * royalty_fee_bps(domain)
+        // TODO: Need to implement Decimals module for calculation
```
Comment (bausano) [inline sources/standards/royalties/royalty_strategy.move:14]: Sam has just implemented something like this for Move, we can wait for that MR to land.

File: sources/trading/bidding.move [hunk 3]
```diff
@@ -161,9 +163,13 @@ module nft_protocol::bidding {
 
         let nft_id = safe::transfer_cap_nft(&transfer_cap);
 
+        // TODO: This function name is confusing, since the
```
Comment (bausano) [inline sources/trading/bidding.move:166]: This can be called by anyone I believe, perhaps `settle_funds`?

File: sources/trading/bidding.move [hunk 4]
```diff
@@ -161,9 +163,13 @@ module nft_protocol::bidding {
...
+        // caller of the function is the seller and not the buyer.
+        // Shall we rename it to receive_funds?
         pay_for_nft<C, FT>(
             &mut bid.offer,
-            bid.buyer,
+            tx_context::sender(ctx),
+            // bid.buyer,
```
Comment (bausano) [inline sources/trading/bidding.move:172]: I need to validate that this is ok

Comment (bausano) [reply sources/trading/bidding.move:172]: This needs to be the buyer, not the tx sender which would be the seller in this case.

**Review by bausano — APPROVED** [review]

File: sources/safe/royalties.move [hunk 5]
```diff
@@ -120,4 +120,12 @@ module nft_protocol::royalties {
...
         });
     }
+
+    // === Getters ===
+
+    public fun amount_u64<C, FT>(
+        trade_payment: &TradePayment<C, FT>,
+    ): u64 {
+        balance::value(&trade_payment.amount)
+    }
```
Comment (Suficio) [inline sources/safe/royalties.move:130]: People can call `balance::value` on the returned `Balance`

```suggestion
    public fun amount<C, FT>(
        trade_payment: &TradePayment<C, FT>,
    ): &Balance<FT> {
        &trade_payment.amount
    }
```

File: sources/standards/royalties/royalty.move (hunk 1 above)
Comment (Suficio) [reply sources/standards/royalties/royalty.move:118]: IMO if the royalty isn't defined then its contribution to the total royalty owed is zero

File: sources/standards/royalties/royalty_strategy.move (hunk 2 above)
Comment (Suficio) [reply sources/standards/royalties/royalty_strategy.move:14]: There are some snippets of code that use floats for this calculation in our codebase already.

This definetly shouldn't be precedent to implement a decimal library as we're dealing with fixed point arithmetic.

**Review by Suficio — CHANGES_REQUESTED** [review]

File: sources/standards/royalties/royalty.move (hunk 1 above)
Comment (nmboavida) [reply sources/standards/royalties/royalty.move:118]: The problem is if the user thinks it has defined the royalties and but it has not, this function will still run, whilst imo it shoudln't

File: tests/utils_2.move [hunk 6]
```diff
@@ -0,0 +1,130 @@
+// TODO: This module is to be merged with test_utils from
```
Comment (bausano) [inline tests/utils_2.move:1]: todo

File: sources/standards/royalties/royalty.move (hunk 1 above)
Comment (nmboavida) [reply sources/standards/royalties/royalty.move:118]: Issue: https://github.com/Origin-Byte/nft-protocol/issues/158

File: sources/trading/bidding.move (hunk 3 above)
Comment (nmboavida) [reply sources/trading/bidding.move:166]: I agree, I renamed it to `settle_funds`

**Review by Suficio — APPROVED** [review]

**Pull request merged**